doc/read-the-docs-site/tutorials/AuctionValidator.hs

```diff
@@ -193,5 +196,5 @@ auctionValidatorScript ::
   CompiledCode (BuiltinData -> BuiltinData -> BuiltinData -> ())
 auctionValidatorScript params =
   $$(PlutusTx.compile [||auctionUntypedValidator||])
-    `PlutusTx.unsafeApplyCode` PlutusTx.liftCode params
+    `PlutusTx.unsafeApplyCode` PlutusTx.liftCode plcVersion100 params
```

Comment:
Idea: Could our plutus ghc plugin be used also to fill in the target-version in the `liftCode`?

The idea would be that:
a) `liftCode` would behave as the current `liftCodeDef` if it was not preprocessed by our plutus ghc plugin.
b) The plutus plugin would transform every  call to `liftCode` it finds to `liftCodeVer passedTargetVersion`.

It would be from one hand more user-friendly, but on the hand more 'magical' thus more error-prone.

Reply:
As an added extra, it could also be controlled behind a boolean plugin flag "addVersionsToLiftCodes"

Reply:
I'm generally anti magic :) I like the fact that at the moment our plugin pretty much only touches stuff that's clearly signalled with a special TH function, it makes it very clear where the cleverness is happening.

The more magical approach that I could be persuaded to consider would be something like promoting `Version` to the type level and having a `KnownVersion` class. Then it could be hidden that way.